rpc/transaction.go

```diff
@@ -218,6 +218,8 @@ type ResourceBounds struct {
 
 // https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L1252
 //
+// Sorry, I think I already asked you why there are *[]*felt.Felt instead of []*felf.Felt, but I can't remember
```

Comment:
because with []*felf.Felt, there is no way to completely omit the field from JSON output. It always gets marshalled to atleast an empty list. *[]*felt.Felt can be entirely omitted if it is nil.